File: Chapter01/Activity1_03/Activity1_03.py
from pyspark.sql import DataFrame
from utilities01_py.helper_python import *


if __name__ == "__main__":
    session: SparkSession = create_session(2, "Zipf validation")
    lines_df: DataFrame = session.read.text(novella_location).withColumnRenamed('value', 'sentences')

    # from pyspark.ml.feature import RegexTokenizer
    # tokenizer = RegexTokenizer(inputCol='sentences', outputCol='words', pattern='\W+')
    #
    # tokenized = tokenizer.transform(lines_df)
    # samples = tokenized.take(100)
    # for sample in samples:
    #     print(sample)

File: Chapter01/Activity1_03/Activity1_03.scala
package Activity1_03


import org.apache.spark.sql.{DataFrame, SparkSession}
import Utilities01.HelperScala.{createSession, novellaLocation}

object Activity1_03 {

  def main(args: Array[String]): Unit = {
    val session: SparkSession = createSession(2, "Spark Tokenizer")
    val linesDf: DataFrame = session.read.text(novellaLocation).withColumnRenamed("value", "sentences")

//    import org.apache.spark.ml.feature.RegexTokenizer
//    val tokenizer = new RegexTokenizer()
//      .setPattern("\\W+")
//      .setInputCol("sentences")
//      .setOutputCol("tokens")
//
//    val tokenized = tokenizer.transform(linesDf)
//    tokenized.take(100).foreach(println(_))

  }

}
